draft-pfeifer-rtgwg-dmpr/draft-pfeifer-rtgwg-dmpr-00.txt

```diff
@@ -187,7 +191,118 @@ Internet-Draft                    DMPR                    September 2017
 
 3.1.  Header
 
-   The binary header description
+3.1.1.  Preamble
+
+   The preamble of a DMPR packet is as follows
+
+    0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |Magic| Reserved|    NextType   |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+                         The preamble of a packet
+
+   Magic
+      A 3-bit Magic: 0b010
+
+   Reserved
+      Reserved for future use
+
+   NextType
+      8-bit unsigned integer.  Defines the type of the immediately
+      following data.  For a NextType of < 128 this data consists an
+      Extension Header.  For a NextType of >= 128 the rest of the packet
+      is payload, the processing is further specified by NextType.
+
+
+
+
+
+
+
+Davies                   Expires March 18, 2018                 [Page 4]
+
+Internet-Draft                    DMPR                    September 2017
+
+
+3.1.2.  Extension Header
+
+   An Extension Header consists of a length specifier, a Data field and
+   a specifier for the type of the immediately following data.
+
+    0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
No, this looks unnatural in the sense this is not the common way to encode extension header. First next type, then length field and then the payload. See following illustration:

![ext header](https://www.cisco.com/c/dam/en/us/td/i/000001-100000/50001-55000/51001-51500/51459.ps/_jcr_content/renditions/51459.jpg)